### ISSUE TITLE
US states are shown on users profile page on Canada

### ISSUE THREAD

**Issue by sergiitk** [DoSomethingArchive/legacy-website#3328]
On Canada [user #1732915](https://canada.dosomething.org/user/1732915/edit) has `CA` (Canada) country in address field:

``` sql
mysql> SELECT field_address_country FROM field_data_field_address WHERE entity_id=1732915;
+-----------------------+
| field_address_country |
+-----------------------+
| CA                    |
+-----------------------+
1 row in set (0.00 sec)
```

However, the address field still shows US states:
![image](https://cloud.githubusercontent.com/assets/672669/4922541/dd8f0f1c-6512-11e4-8c28-536b00903aba.png)

Related issue: #3307
CC @aaronschachter

**Comment by sergiitk** [comment]
Additional: when the form is saved, country field gets reset back to `US`. 
For example, user [user #1732917](https://canada.dosomething.org/user/1732917/edit):

``` sql
> SELECT field_address_country FROM field_data_field_address WHERE entity_id=1732917;
+-----------------------+
| field_address_country |
+-----------------------+
| US                    |
+-----------------------+
1 row in set (0.00 sec)
```

**Comment by aaronschachter** [comment]
This one is nasty. I suspect it's because the `field_address` feature is configured to US:

https://github.com/DoSomething/dosomething/blob/v0.3.96/lib/modules/dosomething/dosomething_user/dosomething_user.features.field_instance.inc#L24

and 

https://github.com/DoSomething/dosomething/blob/v0.3.96/lib/modules/dosomething/dosomething_user/dosomething_user.features.field_instance.inc#L55

This would require altering the feature such that any country be saved, with the disadvantage of needing to add even more `hook_form_alter` fu**ery to make work internationally, saving the site's country code programatically instead of letting Drupal User module do its thing.

Unless we had users just select what country they live in.

**Comment by sergiitk** [comment]
@aaronschachter now I'm experimenting with custom address plugin. Works good so far. I'll fix country tests as described #3393 to make to make testing easier.
